src/coreclr/jit/compiler.cpp

```diff
@@ -7360,9 +7360,9 @@ Compiler::NodeToIntMap* Compiler::FindReachableNodesInNodeTestData()
 
     // Otherwise, iterate.
 
-    for (BasicBlock* block = fgFirstBB; block != nullptr; block = block->bbNext)
+    for (BasicBlock* const block : Blocks())
     {
-        for (Statement* stmt = block->FirstNonPhiDef(); stmt != nullptr; stmt = stmt->GetNextStmt())
+        for (Statement* stmt : block->NonPhiStatements())
         {
             for (GenTree* tree = stmt->GetTreeList(); tree != nullptr; tree = tree->gtNext)
             {
```

Comment:
Nice, the code looks cleaner now! The only missing piece I guess are iterators for GenTrees (LIR and HIR) like in this place.

Reply:
Yeah, we could add an iterator for HIR trees for this case.

LIR already has iterators, e.g. `for (GenTree* node : range)` in `Compiler::gtDispRange`